packages/neon-core/src/rpc/RPCClient.ts

```diff
@@ -239,8 +239,12 @@ export class RPCClient {
   public async getVersion(): Promise<string> {
     try {
       const response = await this.execute(Query.getVersion());
-      const newVersion = response.result.useragent.match(versionRegex)[1];
-      this.version = newVersion;
+      if (response && response.result && response.result.useragent) {
+        const newVersion = response.result.useragent.match(versionRegex)[1];
+        this.version = newVersion;
+      } else {
+        this.version = RPC_VERSION;

```

Comment:
this hides the fact that the response was not to specification. If we are to guard, we should be throwing if `result` or `useragent` is not found.

Reply:
I agree. I will rework and do some more testing. I'm glad we were able to start the discussion and thanks for the help.